synapse/federation/federation_base.py

```diff
@@ -245,3 +249,27 @@ def event_from_pdu_json(pdu_json: JsonDict, room_version: RoomVersion) -> EventB
 
     event = make_event_from_dict(pdu_json, room_version)
     return event
+
+
+def _strip_unsigned_values(pdu_dict: JsonDict) -> JsonDict:
+    """
+    Strip any unsigned values unless specifically allowed, as defined by the whitelist.
+
+    pdu: the json dict to strip values from. Note that the dict is mutated by this
+    function
+    """
+    unsigned = pdu_dict["unsigned"]
+
+    if not isinstance(unsigned, dict):
+        pdu_dict["unsigned"] = {}
+        return pdu_dict
+
+    if pdu_dict["type"] == "m.room.member":
+        whitelist = ["knock_room_state", "invite_room_state", "age"]
+    else:
+        whitelist = ["age"]
+
+    filtered_unsigned = {k: v for k, v in unsigned.items() if k in whitelist}
+    pdu_dict["unsigned"] = filtered_unsigned
+
+    return pdu_dict
```

Comment:
let's remove this: It's no longer used, and I think that returning a value makes it look like we might be returning something different from the input.

```suggestion
```

likewise, s/`return pdu_dict`/`return`/ above, and you'll need to change the annotation to `-> None`.

Reply:
Right makes sense, also thank you for the quick and enlightening review!